src/Tablebot/Plugins/Roll/Plugin.hs

```diff
@@ -57,22 +62,52 @@ rollDice' e' t m = do
     simplify li = li
     countFormatting s = (`div` 4) $ T.foldr (\c cf -> cf + (2 * fromEnum (c == '`')) + fromEnum (c `elem` ['~', '_', '*'])) 0 s
 
+rollDice' :: Maybe (Either ListValues Expr) -> Maybe (Quoted Text) -> ParseUserId -> DatabaseDiscord MessageDetails
+rollDice' e t u@(ParseUserId uid) = do
+  msg <- rollDice'' e t u
+  return
+    ( (messageDetailsBasic msg)
+        { messageDetailsComponents =
+            Just
+              [ ComponentActionRowButton
+                  [ (mkButton buttonName (T.take 100 buttonCustomId))
+                      { componentButtonEmoji = Just (Emoji (Just 0) "🎲" Nothing Nothing Nothing (Just False)),
+                        componentButtonDisabled = buttonDisabled
+                      }
+                  ]
+              ]
```

Comment:
Could maybe be a helper function `makeInteractionButton`, to avoid much use of Nothing? (Again, it might be that there are serious cases where those aren't Nothing and thus my point is not sensible.)

Also, what's with the `T.take`?